### PR TITLE
New geolocation

### PR DESCRIPTION
**Pull request by nboisteault**
I have to improve Lizmap geolocation for a client but I don't want to use old code so I'm creating a geolocation web component which communicates to a Lizmap main object which is a proxy to Lizmap 3. I think we could do that for other parts of LWC in a release_3_5 branch. This way we could remove functionalities old code one by one and create new one the Lizmap 4 way.
We will then make the big jump (but smaller) to Lizmap 4. Lizmap main class will be modified by removing call to Lizmap 3 and adding its own properties for states.

### CODE REVIEW

**Comment by nboisteault** [comment]
@rldhont @laurentj You can now review. @laurentj We'll need node in CI to build JS.

File: lizmap/www/js/edition.js [hunk 1]
```diff
@@ -776,10 +764,13 @@ OpenLayers.Geometry.pointOnSegment = function(point, segment) {
                     // Get SRID and transform geometry
                     var srid = $('#edition-point-coord-crs').val();
                     vertex.transform( editionLayer['ol'].projection,'EPSG:'+srid );
-                    if ( !$('#edition-point-coord-x').attr('disabled') )
+                    if (srid === "4326"){
+                        $('#edition-point-coord-x').val(vertex.x.toFixed(6));
+                        $('#edition-point-coord-y').val(vertex.y.toFixed(6));
+                    }else{
                         $('#edition-point-coord-x').val(vertex.x);
```
Comment (rldhont) [inline lizmap/www/js/edition.js]:
```suggestion
                         $('#edition-point-coord-x').val(vertex.x.toFixed(3));
```

File: lizmap/www/js/edition.js [hunk 2]
```diff
@@ -776,10 +764,13 @@ OpenLayers.Geometry.pointOnSegment = function(point, segment) {
...
-                    if ( !$('#edition-point-coord-y').attr('disabled') )
                         $('#edition-point-coord-y').val(vertex.y);
```
Comment (rldhont) [inline lizmap/www/js/edition.js]:
```suggestion
                         $('#edition-point-coord-y').val(vertex.y.toFixed(3));
```

File: lizmap/www/js/edition.js [hunk 3]
```diff
@@ -835,12 +826,11 @@ OpenLayers.Geometry.pointOnSegment = function(point, segment) {
...
                     $('#edition-point-coord-y').attr('disabled','disabled');
-                    if ( lizMap.controls.geolocation.layer.features.length != 0 ) {
-                        var geometryType = editionLayer.geometryType;
-                        var vertex = lizMap.controls.geolocation.layer.features[0].geometry;
-                        var px = editCtrls[geometryType].handler.layer.getViewPortPxFromLonLat({lon:vertex.x,lat:vertex.y});
-                        editCtrls[geometryType].handler.modifyFeature(px);
-                    }
+
+                    var geometryType = editionLayer.geometryType;
+                    var [lon, lat] = lizMap.mainLizmap.geolocation.getPositionInCRS(editionLayer['ol'].projection);
```
Comment (rldhont) [inline lizmap/www/js/edition.js]: Are you sure that the geolocation return position ?
Are you sure that the geolocation has started ?

File: lizmap/www/js/edition.js (hunk 1 above)
Comment (nboisteault) [reply lizmap/www/js/edition.js]: Are 3 decimals enough for any other cases ?

Comment (rldhont) [reply lizmap/www/js/edition.js]: Can we check if the projection is geographic ?

File: lizmap/www/js/edition.js (hunk 3 above)
Comment (nboisteault) [reply lizmap/www/js/edition.js]: https://github.com/3liz/lizmap-web-client/pull/1450/commits/fd58c8ca7d5dc286e6a2d4d62e6e0c2d647d5af5

**Comment by rldhont** [comment]
@nboisteault thanks

**Comment by nboisteault** [comment]
@rldhont @laurentj it is now finished I think.

**Comment by rldhont** [comment]
You can merge for me.

**Pull request merged**